### PR TITLE
bpo-35325:imp.find_module() return value documentation discrepancy

### PR DESCRIPTION
**Pull request by Windsooon**
I'm not sure we should keep the change in Doc/library/imp.rst, feel free to edit it.

https://bugs.python.org/issue35325

### CODE REVIEW

File: Doc/library/imp.rst [hunk 1]
```diff
@@ -51,7 +51,9 @@ This module provides an interface to the mechanisms used to implement the
    places are searched: the function tries to find a built-in module with the
    given name (:const:`C_BUILTIN`), then a frozen module (:const:`PY_FROZEN`),
    and on some systems some other places are looked in as well (on Windows, it
-   looks in the registry which may point to a specific file).
+   looks in the registry which may point to a specific file). If the path in
```
Comment (ericsnowcurrently) [inline Doc/library/imp.rst]: This paragraph does not need any changes.  This information is already covered below in the paragraph about the return values.

File: Doc/library/imp.rst [hunk 2]
```diff
@@ -66,12 +68,12 @@ This module provides an interface to the mechanisms used to implement the
    contained in the list returned by :func:`get_suffixes` describing the kind of
    module found.
 
-   If the module does not live in a file, the returned *file* is ``None``,
-   *pathname* is the empty string, and the *description* tuple contains empty
-   strings for its suffix and mode; the module type is indicated as given in
-   parentheses above.  If the search is unsuccessful, :exc:`ImportError` is
-   raised.  Other exceptions indicate problems with the arguments or
-   environment.
+   If the module does not live in a file or the path is in the built-in module or
```
Comment (ericsnowcurrently) [inline Doc/library/imp.rst]: While we're here, let's make this paragraph clear by being specific:

```
   If the module is built-in or frozen then *file* and *pathname* are both
   ``None`` and the *description* tuple contains empty...
```

**Review by ericsnowcurrently — CHANGES_REQUESTED** [review]

**Comment by ericsnowcurrently** [comment]
You will want to add a NEWS entry.  The "blurb" tool makes this easy.  [See the devguide](https://devguide.python.org/committing/?highlight=backport#what-s-new-and-news-entries) for more info.

Please be sure to add yourself to Misc/ACKS (in this PR) if you aren't there already.

File: Doc/library/imp.rst (hunk 1 above)
Comment (Windsooon) [reply Doc/library/imp.rst]: Did you mean 
> If the module does not live in a file, the returned file is None, pathname is the empty string,
But it should return None for both file and file_path (not empty string)

**Comment by vstinner** [comment]
I removed the " needs backport to 3.6" label, the 3.6 branch no long accept bugfixes (only security fixes are accepted): https://devguide.python.org/#status-of-python-branches

File: Doc/library/imp.rst (hunk 1 above)
Comment (JulienPalard) [reply Doc/library/imp.rst]: Probably yes, I agree with Eric, let's keep all "return value" information in the same paragraph and keep this paragraph focused only on general behavior.

**Comment by Windsooon** [comment]
I updated the PR based on @ericsnowcurrently 's review. Thank you.

**Pull request merged**